draft-ietf-tls-esni.md

```diff
@@ -1241,16 +1227,85 @@ TCP connections an attacker can open.
-using ECH, the presence of this value will not signal unusual behavior to
-passive eavesdroppers.
+As a means of reducing the impact of network ossification, {{?RFC8744}}
+recommends SNI-protection mechanisms be designed in such a way that network
+operators do not handle connections in which the mechanism is used differently
+than other connections. To that end, ECH is designed to resemble a standard TLS
+handshake as much as possible. The most obvious difference is the extension
+itself: as long as middleboxes ignore it, as required by {{!RFC8446}}, the rest
+of the handshake is designed to look very much as usual. Still, non-compliance
+with established standards is a widespread issue, and it may hinder successful
+deployment of ECH.
+
+The GREASE ECH protocol described in {{grease-ech}} provides a low-risk way to
+evaluate the deployability of ECH on the Internet. It is designed to mimic the
+real ECH protocol ({{real-ech}}), but without actually changing the security
+properties of the handshake. The underlying theory is that if GREASE ECH is
+deployable without triggering differentiable behavior, and real ECH looks enough
+like GREASE ECH, then ECH should be deployable as well. Thus, our strategy for
+mitigating network ossification has two parts:
+1. Design a "cover" protocol (i.e., GREASE ECH) that meets some notion of
+   deployability.
+1. Design the "target" protocol (i.e., ECH) so that networks do not differentiate
+   between it and the cover protocol.
+
+This strategy is analogous to the methodology that lead to "Middlebox
+Compatibility Mode" for TLS 1.3 described in {{!RFC8446}}.
+
+In the first step, the deployability of GREASE ECH is established by deploying
+the protocol and measuring its impact. [[OPEN ISSUE: Run an experiment that
+compares the error rate of GREASE ECH connections with vanilla TLS 1.3.]]
+Viewing the network operator as the adversary for the moment, we model the
+requirements of the second step in terms of the strength of the adversary, as
+follows.
+
+Suppose we know some set of features of the TLS handshake that are known to
+trigger differentiable treatment by the network. This set might include the
+plaintext extensions offered by the client or server, plaintext alert messages,
+the number of round trips over the network, the length of each message, and so
+on. Let us call these the protocol's "observable features". Our goal is to
+design real ECH so that each execution of the protocol has the same observable
+features as if GREASE ECH were being executed instead. For example, if
+extensions are an observable feature, then regardless of whether they are
+running the target protocol or the cover protocol, the client and server should
+appear to negotiate the same set of extensions. That way the real protocol does
+not trigger differentiable treatment that would not have been triggered by the
+cover protocol.
+
+Informally, we will say that target protocol is "indifferentiable" from cover
```

Comment:
```suggestion
Informally, the target protocol is indifferentiable from the cover
```

Reply:
Ack (not taken).

Reply:
My suggestion is to avoid lanaguage like "we say," and rather say, "this specification says." (This isn't a paper after all.)